app/touchnet.js

```diff
@@ -42,7 +42,7 @@ class TouchnetWS {
     let response = await touchnetRequest(this.uri, this.auth, authorizeBody(session_id));
     return {
       receipt:  getSingleNode('/soapenv:Envelope/soapenv:Body/tn:authorizeAccountResponse/tn:creditResponse/tn:receiptNumber', response),
-      user_id:  getSingleNode('/soapenv:Envelope/soapenv:Body/tn:authorizeAccountResponse/tn:ticketName', response),
+      user_id:  getSingleNode('/soapenv:Envelope/soapenv:Body/tn:authorizeAccountResponse/tn:ticketName', response).replace(/_\d+$/, ''),
```

Comment:
In theory, the regex used here is susceptible to regex backtracking issues. In practice, it will never happen because the string involved is one we created and is short. But I won't be surprised if code scanners flag it. It's safe to land, but I might consider a different approach.

Reply:
Never mind. There are no subexpressions here so there is no ReDoS issue.